queries/vala/highlights.scm

```diff
@@ -0,0 +1,387 @@
+; Pointers
+
+(address_of_identifier
+  "&" @symbol
+  (_)*
+)
+
+(pointer_type
+  (_)*
+  "*" @symbol
+)
+
+(indirection_identifier
+  "*" @symbol
+  (_)*
+)
+
+"delete" @keyword
+
+; Misc
+
+(number) @number
+
+[
+  "{"
+  "}"
+  "("
+  ")"
+] @punctuation.bracket
+
+[
+";"
+"."
+","
+"->"
+] @punctuation.delimiter
+
+; Keywords
+
+"return" @keyword.return
+"yield" @keyword.return
+"break" @keyword.return
+
+; Booleans
+
+(true) @boolean
+(false) @boolean
+
+; Operators
+
+(binary_expression
+  (_)
+  [
+    "*"
+    "/"
+    "+"
+    "-"
+    "%"
+    "<"
+    "<="
+    ">"
+    ">="
+    "=="
+    "!="
+    "+="
+    "-="
+    "*="
+    "/="
+    "%="
+    "&&"
+    "||"
+    "&"
+    "|"
+    "^"
+    "~"
+    "|="
+    "&="
+    "^="
+    "??"
+  ] @operator    
+  (_)
+)
+
+(unary_expression
+  (_)?
+  [
+    "-"
+    "!"
+    "--"
+    "++"
+  ] @operator
+  (_)?
+)
+
+(null) @keyword
+
+; Methods
+
+(function_definition
+  (modifier)* @keyword
+  type: (_) @type
+  name: [
+  	(identifier) @method
+    (camel_cased_identifier) @type
+    (generic_identifier (_) @type) 
+    (namespaced_identifier
+    	left: (_)
+        right: (_) @method .
+    )
+  ]
+)
+
+(function_call
+  identifier: [
+  	(identifier) @method
+    (camel_cased_identifier) @type
+    (generic_identifier (_) @type) 
+    (namespaced_identifier
+    	left: (_)
```

Comment:
if you don't need `left` for you query, there's also no need to add it to this file